tests/unit/test_routes.py

```diff
@@ -594,6 +594,47 @@ def test_search(module_client, module_db, fake_auth_from_oc, fake_algolia_save,
     assert (result.json['data'][0]['url'] == resource.json['data'].get('url'))
 
 
+def test_search_paid_filter(module_client, module_db, fake_auth_from_oc, fake_algolia_save, fake_algolia_search):
+    client = module_client
+
+    apikey = get_api_key(client)
+
+    # Create paid and unpaid resource to find it in the search results.
+    paid_resource = client.post("/api/v1/resources",
+                          json=dict(
+                              name=f"Tester1",
+                              category="Website",
+                              url=f"Tester1",
+                              paid=True,
+                          ),
+                          headers={'x-apikey': apikey}
+                          )
+    unpaid_resource = client.post("/api/v1/resources",
+                          json=dict(
+                              name=f"Tester2",
+                              category="Website",
+                              url=f"Tester2",
+                              paid=False,
+                          ),
+                          headers={'x-apikey': apikey}
+                          )
+    
+    # Test on the unpaid resource
+    result = client.get(f"/api/v1/search?paid=false")
+
+    assert (unpaid_resource.status_code == 200)
+    assert (result.status_code == 200)
+    assert (len(result.json["data"]) == 1)
```

Comment:
Since we're mocking the search results with `fake_algolia_search`, it's somewhat pointless to assert on the length of the data. Also it's a problem with tightly coupling the data to the test because if we decided to add a second result to the mock, this would fail even though the endpoint is functioning properly. The only test you have that makes sense to me is calling the endpoint with `paid=true` and asserting a 200 response and then calling the endpoint with `paid=false` and asserting a 200 response.

I'd also like to see some negative unit tests, like calling it with `paid=waffles` or `paid=😊` and asserting that it still returns the results without throwing some kind of 500 error